src/main/java/software/amazon/cloudformation/resource/ResourceTypeSchema.java

```diff
@@ -145,4 +148,48 @@ public static ResourceTypeSchema load(final JSONObject schemaJson) {
     public Map<String, Object> getUnprocessedProperties() {
         return Collections.unmodifiableMap(this.unprocessedProperties);
     }
+
+    public boolean hasWriteOnlyProperties(final JSONObject resourceModel) {
```

Comment:
what's this function for? can't we just strip them if they exist?

Reply:
not used in when we strip them, both other services and packages that consume this may want to know if a model has this without wanting to strip the properties.

Reply:
YAGNI, we can add it later if we need it.